apps/src/tutorialExplorer/tutorialExplorer.js

```diff
@@ -0,0 +1,374 @@
+        </div>
+      );
+    }
+  });
+
+  const FilterSet = React.createClass({
+    propTypes: {
+      filterGroups: React.PropTypes.array,
+      onUserInput: React.PropTypes.func,
+      selection: React.PropTypes.object
+    },
+
+    render() {
+      return (
+        <div>
+          <div className="col-20">
+            <div style={{fontSize: 16}}>
+              Filter By
+            </div>
+            {this.props.filterGroups.map(item => <FilterGroup name={item.name} text={item.text} filterEntries={item.entries} onUserInput={this.props.onUserInput} selection={this.props.selection[item.name]} key={item.name}/>)}
+          </div>
+        </div>
+      );
+    }
+  });
+
+  // For a string of tags, generate a string of their friendly names.
+  function getTagString(prefix, tagString) {
+    if (!tagString) {
+      return "";
+    }
+
+    var tagToString = {
+      length_1_hour: "One hour",
+      subject_english: "English",
+      subject_history: "History",
+      teacher_experience_expert: "Expert",
+      teacher_experience_beginner: "Beginner",
+      student_experience_expert: "Expert",
+      student_experience_beginner: "Beginner",
+      activity_type_programming: "Programming tutorial",
+      grade_pre: "Pre",
+      "grade_2-5": "2-5",
+      "grade_6-8": "6-8",
+      programming_language_javascript: "JavaScript",
+      programming_language_c: "C",
+    };
+
+    return tagString.split(',').map(tag => tagToString[`${prefix}_${tag}`]).join(', ');
+  }
+
+  var TutorialDetail = React.createClass({
+    propTypes: {
+      showing: React.PropTypes.bool,
+      item: React.PropTypes.object,
+      closeClicked: React.PropTypes.func,
+    },
+
+    render: function () {
+      if (!this.props.showing) {
+        // Disable body scrolling.
+        $('body').css('overflow', 'auto');
+        return null;
+      }
+
+      // Enable body scrolling.
+      $('body').css('overflow', 'hidden');
+
+      var tableEntries = [
+        {key: 0, title: "Length",                  body: getTagString("length", this.props.item.tags_length)},
+        {key: 1, title: "Subjects",                body: getTagString("subject", this.props.item.tags_subject)},
+        {key: 2, title: "Educator Experience",     body: getTagString("teacher_experience", this.props.item.tags_teacher_experience)},
+        {key: 3, title: "Student Experience",      body: getTagString("student_experience", this.props.item.tags_student_experience)},
+        {key: 4, title: "Type of Activity",        body: getTagString("activity_type", this.props.item.tags_activity_type)},
+        {key: 5, title: "International Languages", body: getTagString("international_languages", this.props.item.tags_international_languages)},
+      ];
+
+      return (
+        <div id="tutorialPopupFullWidth" style={{position: 'absolute', left: 0, top: 0, width: '100%'}}>
+          <div className="modal" id="tutorialPopup" style={{display: 'block'}} onClick={this.props.closeClicked}>
```

Comment:
In that case, it would be good (if possible) to at least have a generic pegasus dialog component that we could reuse in the future (and have this be a consumer of that dialog). Extra points if that dialog component is similar to the one we use elsewhere, such that one day we can maybe unify.